lib/cache-key.js

```diff
@@ -1,8 +1,14 @@
 'use strict';
 var crypto = require('crypto');
 
+/*
+ * Create a cache key for the given name and dir.
+ *
+ * @param name is the 'require'-style name (i.e., could include a full path,
+ * or be relative to dir)
+ * @param dir optional directory to start resolution from when running
+ * the node-resolution algorithm as 'require' does.
+ */
 module.exports = function cacheKey(name, dir) {
-  var value = name + 0x00 + dir;
-
-  return crypto.createHash('sha1').update(value).digest('hex');
+  return crypto.createHash('sha1').update(name + '|' + dir).digest('hex');
```

Comment:
should we use `"\x00"` instead of '|' ? So we have a null char, and no accidental collisions?

Reply:
I don't think so, because 0x00 is not a null character, it's converted from numeric 0 to a '0' string when inserted into the string. That's much worse than |, IMO.

Reply:
Ya, i typo'd it by mistake [more](https://github.com/stefanpenner/hash-for-dep/pull/45#discussion_r)

Reply:
I switched to using '\x00'.